recipe/install.py

```diff
@@ -0,0 +1,188 @@
+            )
+        )
+    )
+    excluded = set(
+        itertools.chain(
+            *(
+                glob.glob(
+                    str(STAGE / pathlib.Path(item)),
+                    recursive=True,
+                )
+                for item in exclude
+            )
+        )
+    )
+    for match in sorted(included - excluded):
+        match = pathlib.Path(match)
+        if match.exists():
+            relative = match.relative_to(STAGE)
+            if not match.is_dir():
+                print(relative)
+                os.makedirs((PREFIX / relative).parent, exist_ok=True)
+                shutil.copy(
+                    match,
+                    PREFIX / relative,
+                    follow_symlinks=False,
+                )
+
+
+def sort_artifacts_based_on_name(basename):
+    PKG_NAME = os.environ["PKG_NAME"]
+
+    print(f"Installing {PKG_NAME} to {PREFIX} for {target_platform}")
+    print("Based on the package name, ", end="")
+
+    if re.match(r"^.+\-split$", PKG_NAME):
+        raise ValueError("The top level package should not run this script.")
+
+    # libfoo OR foo-dev OR libfoo-dev
+    # dav1d-dev, v8-dev, m-dev, secp256k1-dev
+    # libdav1d, libv8, libm, libsecp256k1
+    # libdav1d-dev, libv8-dev, libm-dev, libsecp256k1-dev
+    if (
+        re.match(f"^lib{ basename }$", PKG_NAME)
+        or re.match(f"^{ basename }-dev$", PKG_NAME)
+        or re.match(f"^lib{ basename }-dev$", PKG_NAME)
+    ):
+        print("this package is needed for compiling/linking.")
+        glob_install(
+            include=[
+                "include/**",
+                "lib/**",
+            ],
+            exclude=[
+                # versioned libs
+                "lib/lib*.*.dylib",
+                "lib/lib*.so.*",
+                # static libs
+                "lib/lib*.a",
+                "lib/lib*.lib",
+                "lib/*.a.lib",
+                "lib/*_a.lib",
+                "lib/*static.lib",
+                "lib/**/*static*",
+            ],
+        )
+        return
+
+    # libfoo1
+    # libdav1d1, libv8-1, libm1, libsecp256k1-1
+    if re.match(f"^lib{ basename }-?[0-9]+$", PKG_NAME):
+        print("this package is versioned so/dylib, dlls.")
+        glob_install(
+            include=[
+                "bin/*.dll",
+                "lib/lib*.*.dylib",
+                "lib/lib*.so.*",
+            ]
+        )
+        return
+
+    # foo
+    # dav1d, v8, m, secp256k1
+    if re.match(f"^{ basename }$", PKG_NAME):
+        print("this package is tools, docs, and misc files needed for tools.")
+        glob_install(
+            include=[
+                "bin/**",
+                "doc/**",
+                "share/**",
+            ],
+            exclude=[
+                "bin/*.dll",
+            ],
+        )
+        return
+
+    # libfoo-static
+    # libdav1d-static, libv8-static, libm-static, libsecp256k1-static
+    if re.match(f"^lib{ basename }-static$", PKG_NAME):
```

Comment:
```suggestion
    if "lib{ basename }-static" == PKG_NAME:
```
keep it simple?

Reply:
I agree, for the non-complicated pattern matches, I should probably just use equality instead of regex.